inference-engine/src/mkldnn_plugin/nodes/base.hpp

```diff
@@ -54,6 +56,61 @@ class ExtLayerBase: public ILayerExecImpl {
     }
 
 protected:
+    class DataConfigurator {
+    public:
+        explicit DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType):
+                tensorDescType(tensorDescType) {}
+
+        DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType, Precision prc):
+                tensorDescType(tensorDescType), prc(prc) {}
+
+        DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType, Precision prc, bool constant, int inplace = -1):
+                tensorDescType(tensorDescType), prc(prc), constant(constant), inplace(inplace) {}
+
+        const MKLDNNPlugin::TensorDescCreatorTypes tensorDescType;
```

Comment:
M.b. it would be better to store `TensorDescCreator::CreatorConstPtr` instead of the enumeration member and access the creator directly in the `addConfig() ` call. And it might also make sense to add another constructor that will receive such a creator pointer to allow the user to use a custom creator if he needs to, e.g.
```cpp
   DataConfigurator( const TensorDescCreator::CreatorConstPtr& creator, Precision prc = UNSPECIFIED) ;
```
And the constructor from `MKLDNNPlugin::TensorDescCreatorTypes tensorDescType` still remains, it just initializes the CreatorConstPtr from the common creators map. Also the checks on lines 91-93 could be moved there.

Reply:
Good suggestion. Done